app/experimenter/projects/forms.py

```diff
@@ -23,8 +23,11 @@ def clean_name(self):
     def clean(self):
         cleaned_data = super().clean()
 
-        name = cleaned_data.get("name")
-        cleaned_data["slug"] = slugify(name)
+        if self.instance.slug:
+            del cleaned_data["slug"]
+        else:
+            name = cleaned_data.get("name")
+            cleaned_data["slug"] = slugify(name)
```

Comment:
This might be outside the scope of this PR, but why are we doing this in the form code and not in the model code?

Reply:
@glasserc Putting this automation on the model makes it difficult to edit it by hand from the admin or override for testing in a console or in unit tests.  

In general I try to avoid putting too much data automation in models because for every place it makes your life easier it'll blow up in some annoying way somewhere else, whereas keeping it in the form makes it clear that from the user facing workflow it'll take place, but if you go in under the hood you shouldn't expect any 'magic'.